examples/Examples/Field.hs

```diff
@@ -304,4 +304,4 @@ tripYZComp :: String
 tripYZComp = "(program 1.0.0 (\\i0 -> (\\i0 -> addInteger (unIData (force headList i1)) (unIData (force headList (force tailList i1)))) (force tailList ((\\i0 -> force (force sndPair) (unConstrData i1)) i1))))"
 
 letSomeFieldsComp :: String
-letSomeFieldsComp = "(program 1.0.0 (\\i0 -> (\\i0 -> (\\i0 -> (\\i0 -> (\\i0 -> addInteger (addInteger (unIData (force headList i4)) (unIData (force headList i3))) (unIData (force headList (force tailList i1)))) (force tailList i1)) (force tailList i1)) (force tailList i1)) ((\\i0 -> force tailList (force tailList (force tailList i1))) i1)))"
+letSomeFieldsComp = "(program 1.0.0 (\\i0 -> (\\i0 -> (\\i0 -> addInteger (addInteger (unIData (force headList i2)) (unIData (force headList i1))) (unIData (force headList (force tailList (force tailList (force tailList i1)))))) (force tailList i1)) (force tailList (force tailList (force tailList i1)))))"
```

Comment:
I just noticed that there is an inefficiency that can't be solved easily, but we can make it better. The `force headList x` part should be plet'ed.

Reply:
It's still not perfect, because if you use it only once it should be inlined. There's not really a good way of doing that, but it's better that it be a plet IMO. It's very unintuitive for the field access to be expensive IMO. Maybe there's some other way we can make users `plet` it themselves.

Reply:
This could be somewhat problematic since, very often - people use a field only once. In which case, this would be a rather unnecessary `plet`. But I absolutely understand the concern.

Reply:
On a related note: what's up with so much builtin duplication in that compiled output? Is that missing hoisting?

Reply:
It would be a simple change to `plet` bind all the `force headList x` terms by default. The question is whether we want that.

I think it's acceptable enough for users to have to know to plet bind the terms if they're going to use them more than once, aside from changes to hoisting I'm not sure if we can have it work automagically

Reply:
We are in fact missing some `phoistAcyclic` here! Good catch. #177

Reply:
This is very odd, I can't find what's not hoisted.

Reply:
#195

Reply:
It's ok - I'll be testing these parts pretty extensively in the coming days, hopefully we'll be able to catch everything one by one.